edgelet/edgelet-kube/src/convert/to_k8s.rs

```diff
@@ -1,11 +1,673 @@
+            if element_count >= 2 {
+                let bind_name = sanitize_dns_value(bind_elements[0])?;
+                let host_path_volume_source = api_core::HostPathVolumeSource {
+                    path: bind_elements[0].to_string(),
+                    type_: Some("DirectoryOrCreate".to_string()),
+                };
+                let bind_volume = api_core::Volume {
+                    name: bind_name.clone(),
+                    host_path: Some(host_path_volume_source),
+                    ..api_core::Volume::default()
+                };
+
+                let bind_mount = api_core::VolumeMount {
+                    mount_path: bind_elements[1].to_string(),
+                    name: bind_name,
+                    read_only: Some(element_count > 2 && bind_elements[2].contains("ro")),
+                    ..api_core::VolumeMount::default()
+                };
+
+                volumes.push(bind_volume);
+                volume_mounts.push(bind_mount);
+            } else {
+                warn!("Bind: bind mount did not follow format source:target[:ro|rw]");
+            }
+        }
+    };
+    if let Some(mounts) = spec
+        .config()
+        .create_options()
+        .host_config()
+        .as_ref()
+        .and_then(|hc| hc.mounts())
+    {
+        for mount in mounts.iter() {
+            match mount._type() {
+                Some("bind") => {
+                    if let (Some(source), Some(target)) = (mount.source(), mount.target()) {
+                        let bind_name = sanitize_dns_value(source)?;
+
+                        let host_path_volume_source = api_core::HostPathVolumeSource {
+                            path: source.to_string(),
+                            type_: Some("DirectoryOrCreate".to_string()),
+                        };
+                        let bind_volume = api_core::Volume {
+                            name: bind_name.clone(),
+                            host_path: Some(host_path_volume_source),
+                            ..api_core::Volume::default()
+                        };
+
+                        let bind_mount = api_core::VolumeMount {
+                            mount_path: target.to_string(),
+                            name: bind_name,
+                            read_only: mount.read_only().cloned(),
+                            ..api_core::VolumeMount::default()
+                        };
+
+                        volumes.push(bind_volume);
+                        volume_mounts.push(bind_mount);
+                    } else {
+                        warn!("Bind mount did not contain a source and target");
+                    }
+                }
+                Some("volume") => {
+                    if let (Some(source), Some(target)) = (mount.source(), mount.target()) {
+                        let volume_name = sanitize_dns_value(source)?;
+
+                        let volume = if runtime.use_pvc() {
```

Comment:
Maybe some comments here to describe what's going on might help (with the the `use_pvc` check)?

Reply:
Will do.